packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -315,16 +341,54 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         finalizedWithdrawals[withdrawalHash] = true;
 
         // Set the l2Sender so contracts know who triggered this withdrawal on L2.
+        // This acts as a reentrancy guard.
         l2Sender = _tx.sender;
 
-        // Trigger the call to the target contract. We use a custom low level method
-        // SafeCall.callWithMinGas to ensure two key properties
-        //   1. Target contracts cannot force this call to run out of gas by returning a very large
-        //      amount of data (and this is OK because we don't care about the returndata here).
-        //   2. The amount of gas provided to the execution context of the target is at least the
-        //      gas limit specified by the user. If there is not enough gas in the current context
-        //      to accomplish this, `callWithMinGas` will revert.
-        bool success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, _tx.value, _tx.data);
+        bool success;
+        (address token,) = gasPayingToken();
+        if (token == Constants.ETHER) {
+            // Trigger the call to the target contract. We use a custom low level method
+            // SafeCall.callWithMinGas to ensure two key properties
+            //   1. Target contracts cannot force this call to run out of gas by returning a very large
+            //      amount of data (and this is OK because we don't care about the returndata here).
+            //   2. The amount of gas provided to the execution context of the target is at least the
+            //      gas limit specified by the user. If there is not enough gas in the current context
+            //      to accomplish this, `callWithMinGas` will revert.
+            success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, _tx.value, _tx.data);
+        } else {
+            // Cannot call the token contract directly from the portal. This would allow an attacker
+            // to call approve from a withdrawal and drain the balance of the portal.
+            if (_tx.target == token) revert BadTarget();
+
+            // Only transfer value when a non zero value is specified. This saves gas in the case of
+            // using the standard bridge or arbitrary message passing.
+            if (_tx.value != 0) {
+                // Read the balance of the target contract before the transfer so the consistency
+                // of the transfer can be checked afterwards.
+                uint256 balanceOf = IERC20(token).balanceOf(address(this));
+
+                // Update the contracts internal accounting. Do not use the scaled value as this
+                // is tracking the number of units that are minted in the L2.
+                _balance -= _tx.value;
```

Comment:
I  believe the "scaled value" comment is outdated since we have an 18 decimal (`GAS_PAYING_TOKEN_DECIMALS`) requirement

Reply:
Ah good catch